161967142  Estimate article read time

Darius-Ndubi opened the pull request:
## What does this PR do?
 - Estimates the amount of time it will take a user to read an article

## Description of Task to be completed?
- Adds a field to show how long it takes to read a comment
-  Adds test on estimating article read time

## How should this be manually tested?

- Clone the repo and cd into the project folder
- Create a virtual environment by running `$ python3 -m venv <name_of_environment>`
- Activate the virtual environment by running `source <name_of_environment>/bin/activate`
- Run `$ pip install -r requirements.txt`
- Then add the following environment variables to your computer.

  - `$ export DB_NAME='choose_a_dbname'`
  - `$ export DB_USER='db_username'`
  - `$ export DB_PASSWORD='db_role_password'`
  - `$ export DB_HOST='localhost or preffered host'`
  - `$ export DB_PORT='server_port_number'`
  - `$ export APP_DEBUG='True' `
  - `$ export SOCIAL_AUTH_FACEBOOK_KEY='<facebook_app_id>'`
  - `$ export SOCIAL_AUTH_FACEBOOK_SECRET='<facebook_app_secret>'`
  - `$ export SOCIAL_AUTH_TWITTER_KEY='<twitter_api_key>'`
  - `$ export SOCIAL_AUTH_TWITTER_SECRET='<twitter_api_secret>'`
  - `$ export SOCIAL_AUTH_GOOGLE_OAUTH2_KEY='<google_client_id>'`
  - `$ export SOCIAL_AUTH_GOOGLE_OAUTH2_SECRET='<google_client_secret>'`

- Change the current branch to ft-estimate-article-read-time-161967142
-  Run `python3 manage.py makemigrations`
-  Run `python3 manage.py migrate`
-  Run `python3 manage.py test`

## What are the relevant pivotal tracker stories?
[#161967142](https://www.pivotaltracker.com/story/show/161967142)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

## Screenshots 
<img width="979" alt="screenshot 2018-12-18 at 18 48 01" src="https://user-images.githubusercontent.com/16937341/50171427-7852fa80-0303-11e9-9b1e-97bfb54b7e78.png">

bettblake08 reviewed (approved):
👍

Bryan-Cee reviewed (approved):
👍

Bryan-Cee reviewed (approved):
👍

bettblake08 commented on authors/apps/articles/serializers.py:
```diff
@@ -28,6 +30,16 @@ def validate_user_permissions(self, request, data):
+
+        total_words = data.get('description') + ' ' + data.get('body')
+
+        time_to_read = len(total_words.split()) / 180
```
Add a function to strip html tags. This is because we need to calculate the time to read the words only, exclude html tags.

Darius-Ndubi replied:
@bettblake08  I am working in that will update that method. Thanks for spotting that.

Darius-Ndubi replied:
@bettblake08 it's now updated, the <> in the body will be removed when estimating time to read an article.

bettblake08 replied:
@Darius-Ndubi It's good.

Darius-Ndubi committed:
feat(estimate time) : Estimate read time for an article

 - Add article read time estimation
 - Add tests on article read time

[Delivers #161967142

bettblake08 reviewed (approved):
👍

mutoromike reviewed (approved):
LGTM
